### PR TITLE
Move IClientErrorActionResult to 4xx status code results

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace Microsoft.AspNetCore.Mvc
     /// Represents an <see cref="ActionResult"/> that when executed will
     /// produce an HTTP response with the given response status code.
     /// </summary>
-    public class StatusCodeResult : ActionResult, IClientErrorActionResult
+    public class StatusCodeResult : ActionResult, IStatusCodeActionResult
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs]: This was weird - not all status code results are client errors. Moving the interface to the 4xx ones

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs [hunk 2]
```diff
@@ -288,7 +288,7 @@ static MvcCoreLoggerExtensions()
             _httpStatusCodeResultExecuting = LoggerMessage.Define<int>(
                 LogLevel.Information,
                 1,
-                "Executing HttpStatusCodeResult, setting HTTP status code {StatusCode}");
+                "Executing StatusCodeResult, setting HTTP status code {StatusCode}");
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs]: Some logging typos

**Comment by pranavkm** [comment]
FYI @ryno

**Comment by pranavkm** [comment]
@dotnet-bot build OSX 10.12 Release Build please

**Comment by rynowak** [comment]
What does this do? were these not getting converted to problem details?

**Comment by pranavkm** [comment]
They were doing fine. It just meant that things like `OkResult` would now qualify for `ProblemDetails` which seemed somewhat misleading.

**Comment by pranavkm** [comment]
I'll just document the `IClientErrorActionResult` a bit better. This isn't necessary.

**Pull request closed without merging**